src/networking.c

```diff
@@ -1667,7 +1668,8 @@ void freeClient(client *c) {
                                   REDISMODULE_SUBEVENT_REPLICA_CHANGE_OFFLINE,
                                   NULL);
         if (c->flags & CLIENT_REPL_RDB_CHANNEL) {
-            dictDelete(server.slaves_waiting_psync, &c->id);
+            c->flags &= ~CLIENT_PROTECTED;
```

Comment:
why do we need to clear the flag here?

Reply:
While I was writing it, I thought COB overruns would be handled here during the grace period. I now believe that we won't be able to reach this point under such circumstances. 
For now, I will remove it, and then determine the COB overrun case once I have covered it with tcl tests.